lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -410,6 +428,9 @@ LogicalResult CppEmitter::emitAttribute(Attribute attr) {
     return success();
   }
   if (auto dense = attr.dyn_cast<mlir::DenseFPElementsAttr>()) {
+    // Dense attributes are not supported if emitting C
+    if (restrictedToC())
+      return failure();
```

Comment:
Would be good to emit a descriptive error message here, but I don't think that would be properly propagated by the caller at the moment. I'd suggest to address this in a follow up and add tests for it if possible (see [here](https://mlir.llvm.org/getting_started/TestingGuide/#diagnostic-verification-tests)).

Reply:
At the moment, the `CppEmitter` only returns a `LogicalResult`. But I agree, a more descriptive error message would be desirable.